### PR TITLE
Async pipe can now handle StreamController.stream's properly

### PR DESCRIPTION
**Pull request by daniel-v**
fixes dart-lang/angular2#260

I imagine a couple of test cases could be added to this fix.

### CODE REVIEW

File: lib/src/common/pipes/async_pipe.dart [hunk 1]
```diff
@@ -69,7 +69,8 @@ class AsyncPipe implements OnDestroy {
       this._latestReturnedValue = this._latestValue;
       return this._latestValue;
     }
-    if (!identical(obj, this._obj)) {
+    print('${_maybeStreamIdentical(obj, this._obj)}');
```
Comment (matanlurey) [inline lib/src/common/pipes/async_pipe.dart]: Remove print.

File: lib/src/common/pipes/async_pipe.dart [hunk 2]
```diff
@@ -69,7 +69,8 @@ class AsyncPipe implements OnDestroy {
...
+    if(!_maybeStreamIdentical(obj, this._obj)) {
```
Comment (matanlurey) [inline lib/src/common/pipes/async_pipe.dart]: Add a one/two line comment on why this is necessary for future devs.

File: lib/src/common/pipes/async_pipe.dart [hunk 3]
```diff
@@ -114,4 +115,12 @@ class AsyncPipe implements OnDestroy {
       this._ref.markForCheck();
     }
   }
+
+  static bool _maybeStreamIdentical(dynamic a, dynamic b) {
```
Comment (matanlurey) [inline lib/src/common/pipes/async_pipe.dart]: Can be a little cleaner:

```dart
static bool _maybeStreamIdentical(a, b) {
  if (!identical(a, b)) {
    return a is Stream && b is Stream && a == b;
  }
  return true;
}
```

**Comment by daniel-v** [comment]
Here is an example that did not work for me before: https://gist.github.com/daniel-v/4404bfceff16a6f8c68a9a295474fadd

There is a little more info in #260 .

**Comment by daniel-v** [comment]
Could I get an update on this?

**Comment by matanlurey** [comment]
Reminding myself to get this in tomorrow AM. Sorry about that.

**Pull request closed without merging**